main.nf

```diff
@@ -78,24 +79,21 @@ process BuildDockerContainers {
 dockerContainersBuilt = dockerContainersBuilt.view {"Docker container: $repository/$it:$tag built."}
 
 process BuildSingularityContainers {
-  tag {container}
+  tag {repository + "/" + container + ":" + tag}
+
+  publishDir publishSingularityDir, mode: 'copy'
 
   input:
     val container from singularityContainers
 
   output:
-    val container into singularityContainersBuilt
+    file("*.img") into singularityContainersBuilt
 
   when: singularity
 
   script:
   """
-  docker run \
-  -v /var/run/docker.sock:/var/run/docker.sock \
-  -v /tmp/$container:/output \
-  --privileged -t --rm \
-  singularityware/docker2singularity \
-  $repository/$container:$tag
+  singularity pull --name $container-${tag}.img docker://$repository/$container:$tag
```

Comment:
I had like so many containers, I had to make a script to manage that, and since we're already using Nextflow, I figured it was good to hone my skills.
I would have been a shame to make a Makefile for it ;-)

And yes, this is the singularity config file.